### PR TITLE
Added TypeScript definitions and fileSize attribute in `createReadStream`

### PR DESCRIPTION
**Pull request by pkeuter**
Thanks for this very useful library! I've added some TypeScript definitions which you might find useful. For range-request related calls, I've also added a `fileSize` property to the `readableStream` object.

### CODE REVIEW

File: lib/api/createReadStream.js [hunk 1]
```diff
@@ -33,6 +34,7 @@ module.exports = function createReadStream(path, options, cb) {
     var close = request.bind(undefined, 'close', file, connection);
 
     var stream = new Readable();
+    stream.fileSize = fileSize;
```
Comment (julien-f) [inline lib/api/createReadStream.js:37]: Is it standard?

Are there some widely used streams that used such a property?

Comment (pkeuter) [reply lib/api/createReadStream.js:37]: No it's not standard, we could add it as a third parameter in the callback as well if you prefer that? This was just an easy way for me to get the fileSize.

Comment (julien-f) [reply lib/api/createReadStream.js:37]: I was just wondering about the name of the property but `fileSize` is good enough.

**Comment by julien-f** [comment]
Please split this PR: one for `fileSize` and another one for typedefs.

**Comment by pkeuter** [comment]
@julien-f  done.

**Pull request closed without merging**